libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/IndividualFeatureImportanceView/LargeIndividualFeatureImportanceView/LargeIndividualFeatureImportanceView.tsx

```diff
@@ -0,0 +1,292 @@
+          onWeightChange={this.props.onWeightChange}
+          weightOptions={this.props.weightOptions}
+          weightLabels={this.props.weightLabels}
+          modelType={this.props.modelType}
+        />
+      </Stack>
+    );
+  }
+
+  private setIsRevertButtonClicked = (status: boolean): void => {
+    this.setState({ isRevertButtonClicked: status });
+  };
+
+  private updateBubblePlotData = async (
+    chartProps: IGenericChartProps
+  ): Promise<void> => {
+    this.setState({
+      indexSeries: [],
+      isBubbleChartDataLoading: true,
+      isLocalExplanationsDataLoading: false,
+      localExplanationsData: undefined,
+      localExplanationsErrorMessage: undefined,
+      selectedPointsIndexes: [],
+      xSeries: [],
+      ySeries: []
+    });
+    const datasetBarConfigOverride = await getBubblePlotData(
+      chartProps,
+      this.props.cohort,
+      this.context.jointDataset,
+      this.context.dataset,
+      this.state.isLocalExplanationsDataLoading,
+      this.context.requestBubblePlotData,
+      this.selectPointFromChartLargeData,
+      this.onBubbleClick
+    );
+    if (
+      datasetBarConfigOverride &&
+      !instanceOfHighChart(datasetBarConfigOverride)
+    ) {
+      this.setState({
+        bubbleChartErrorMessage: getErrorMessage(datasetBarConfigOverride),
+        highChartConfigOverride: undefined,
+        isBubbleChartDataLoading: false
+      });
+      return;
+    }
+    this.setState({
+      chartProps,
+      highChartConfigOverride: datasetBarConfigOverride,
+      isBubbleChartDataLoading: false,
+      isBubbleChartRendered: true,
+      isRevertButtonClicked: false
+    });
+  };
+
+  private updateScatterPlotData = (chartProps: IGenericChartProps): void => {
+    const datasetBarConfigOverride = getScatterOption(
+      this.state.xSeries,
+      this.state.ySeries,
+      this.state.indexSeries,
+      chartProps,
+      this.context.jointDataset,
+      this.state.selectedPointsIndexes,
+      [],
+      this.state.isLocalExplanationsDataLoading,
+      true,
+      false,
+      this.selectPointFromChartLargeData
+    );
+    this.setState({
+      chartProps,
+      highChartConfigOverride: datasetBarConfigOverride,
+      isBubbleChartRendered: false,
+      isRevertButtonClicked: false
+    });
+  };
+
+  private onBubbleClick = (
+    scatterPlotData: IHighchartsConfig,
+    xSeries: number[],
+    ySeries: number[],
+    indexSeries: number[]
+  ): void => {
+    this.setState({
+      highChartConfigOverride: scatterPlotData,
+      indexSeries,
+      isBubbleChartRendered: false,
+      xSeries,
+      ySeries
+    });
+  };
+
+  private onXSet = (value: ISelectorConfig): void => {
+    if (this.state.chartProps) {
+      const newProps = getNewChartProps(value, true, this.state.chartProps);
+      this.setState({ chartProps: newProps });
+    }
+  };
+
+  private onYSet = (value: ISelectorConfig): void => {
```

Comment:
the only difference for onXSet and onYSet is `xSet: boolean`, maybe we can combine these two methods into one by passing `xSet: boolean`

Reply:
onXSet and onYset is required by onAccept of AxisConfig. As two AxisConfig is used, we need two different functions to handle. Please let me know if there is a way to merge this. Happy to merge.